src/flownet/utils/plot_results.py

```diff
@@ -0,0 +1,372 @@
+import argparse
+import pathlib
+import re
+from datetime import datetime
+from typing import Optional, List
+
+import matplotlib
+import matplotlib.pyplot as plt
+import pandas as pd
+from fmu import ensemble
+from ecl.summary import EclSum
+
+from .observations import _read_ert_obs
+
+matplotlib.use("Agg")
+
+
+def plot_ensembles(
+    ensemble_type: str,
+    vector: str,
+    ensembles_data: List[pd.DataFrame],
+    plot_settings: dict,
+):
+    """Function to plot a list of ensembles.
+
+    Args:
+        ensemble_type: prior or posterior
+        vector: Name of the vector to plot
+        ensembles_data: List of dataframes with ensemble data
+        plot_settings: Settings dictionary for the plots.
+
+    Returns:
+        Nothing
+
+    Raises:
+        Value error if incorrect plot type.
+
+    """
+    if not ensemble_type in ("prior", "posterior"):
+        raise ValueError("Plot type should be either prior or posterior.")
+
+    for i, ensemble_data in enumerate(ensembles_data):
+
+        ensemble_data = (
+            remove_duplicates(ensemble_data[["DATE", "REAL", vector]])
+            .pivot(index="DATE", columns="REAL", values=vector)
+            .dropna()
+        )
+
+        color = (
+            plot_settings[f"{ensemble_type}_colors"][0]
+            if len(plot_settings[f"{ensemble_type}_colors"]) == 1
+            else plot_settings[f"{ensemble_type}_colors"][i]
+        )
+        alpha = (
+            plot_settings[f"{ensemble_type}_alphas"][0]
+            if len(plot_settings[f"{ensemble_type}_alphas"]) == 1
+            else plot_settings[f"{ensemble_type}_alphas"][i]
+        )
+
+        plt.plot(
+            ensemble_data.index,
+            ensemble_data.values,
+            color=color,
+            alpha=alpha,
+            linestyle="solid",
+        )
+
+
+def plot(
+    vector: str,
+    prior_data: list,
+    posterior_data: list,
+    reference_simulation: Optional[EclSum],
+    plot_settings: dict,
+):
+    """Main plotting function that generate builds up a single plot build up
```

Comment:
A bit hard to read... :)

```suggestion
    """Main plotting function that generate a single plot from potentially multiple ensembles and other data
```

or something